pg_shard.c

```diff
@@ -246,15 +256,26 @@ PgShardPlanner(Query *query, int cursorOptions, ParamListInfo boundParams)
 		if (selectFromMultipleShards)
 		{
 			Oid distributedTableId = InvalidOid;
+			Query *localQuery = NULL;
+			List *queryRestrictList = QueryRestrictList(distributedQuery);
+			List *remoteRestrictList = NIL;
+			List *localRestrictList = NIL;
+
+			/* TODO: replace with call to classify restrictions */
+			remoteRestrictList = queryRestrictList;
+
+			/* build local and distributed query */
+			distributedQuery = RowAndColumnFilterQuery(distributedQuery,
```

Comment:
Hm. I honestly don't know. The problem is the object is a `DistributedPlan`, so there's a duality in that we build the `distributedPlan` using the `distributedQuery` but use the `localQuery` to build a local plan (the sequential scan, currently just named `plannedStatement`). I'll have to think a bit more about this.